mysql-test/suite/tianmu/t/issue845.test

```diff
@@ -0,0 +1,22 @@
+--source include/have_tianmu.inc
+
+USE test;
+
+--disable_warnings
+
+DROP TABLE IF EXISTS test_timestamp;
+
+CREATE TABLE test_timestamp (t timestamp);
+INSERT INTO test_timestamp VALUES ('2019-01-02 00:01:02'),
+                              ('2019-01-02 01:02:03'),
+                              ('2019-01-02 02:03:04'),
+                              ('2019-01-02 03:05:06'),
+                              ('2019-01-02 04:07:08');
+
+#EXTRACT SELECT
+
+SELECT EXTRACT(DAY_HOUR FROM t), EXTRACT(MINUTE_SECOND FROM t) FROM test_timestamp;
```

Comment:
adds some cases  just as following:
```
mysql> SELECT EXTRACT(YEAR FROM '2019-07-02');
        -> 2019
mysql> SELECT EXTRACT(YEAR_MONTH FROM '2019-07-02 01:02:03');
        -> 201907
mysql> SELECT EXTRACT(DAY_MINUTE FROM '2019-07-02 01:02:03');
        -> 20102
mysql> SELECT EXTRACT(MICROSECOND
    ->                FROM '2003-01-02 10:30:00.000123');
        -> 123
```

Reply:
we should be aware of corner cases. 

[EXTRACT(unit FROM **date**)](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_extract)

```
Returns NULL if date is NULL.
```

Reply:
add cases to  contain these conditions